### DISCUSSION TITLE
Is it possible in the plugin code to get the value of cli-argument '--retry-streams'?

### DISCUSSION THREAD

**Discussion by simvesel** [streamlink/streamlink#5392] [Q&A] [answered]
### Description

Hello!
**Is it possible in the plugin code to get the value of cli-argument '--retry-streams'?**

_**Situation:**_ streamlink works in "sleep mode" with a delay of 2-10 minutes between repeated requests.
If an error occurred during the request to get the URL playlist.m3u8 (an http-timeout worked or the server was restarted), then I would like to retry the HTTP request to the server.
But I wouldn't to handle errors in this way in the plugin code all the time, when live-stream capture starts, shell-script execute streamlink with a small delay (5-10 seconds) and then there is no need to make a repeated HTTP request in case of an error.
Therefore, I would like to receive the value '--retry-streams' in the plugin code in order to change the error handling behavior or implement some special behavior for low latency cases.

**Comment by bastimeyer** [comment]
Plugins only have access to session options. `--retry-streams` is none of those and just a value used by the streamlink_cli package.
https://streamlink.github.io/api.html#streamlink.Streamlink.set_option

I don't understand what you're trying to explain. You can write a simple shell script which checks for the stream availability and which then starts playing/recording right away when a stream is available, without any further delay.

**Comment by simvesel** [comment]
Thanks for the answer.
Unfortunately, I would like to do this only through the plugin, and not also write the shell script code, which also checks for the existing of playlist.m3u8. The plugin also will do this (there will be duplication of functionality).

**Reply by bastimeyer** [reply, marked as answer]
It is still not clear what you're trying to do.

The purpose of a plugin is to return a dict of `Stream` objects in its `_get_streams()` method. If that dict is empty because no available streams were found by the plugin, then `streamlink_cli` will try to repeat the query if `--retry-streams` is set.

- https://github.com/streamlink/streamlink/blob/5.5.1/src/streamlink_cli/main.py#L531-L545
- https://github.com/streamlink/streamlink/blob/5.5.1/src/streamlink_cli/main.py#L440-L476
- https://github.com/streamlink/streamlink/blob/5.5.1/src/streamlink/plugin/plugin.py#L421-L435

The only condition here is that the plugin doesn't raise `FatalPluginError` or any kind of other exception which isn't a subclass of `PluginError`. If a plugin does that, then `--retry-streams` won't work, but that is intended behavior. So if you're writing your own plugin, then make sure that it doesn't raise errors when no streams are available.

**Reply by simvesel** [reply]
Thank you very much for detail explain! 
I thought that it is not necessary to complicate the plugin code in such a way that it handles errors HTTP requests differently when requesting to get the playlist.m3u8 URL, when `--retry-streams` has a small value. It's easier to implement the one error handling behavior on HTTP requests in the plugin.
